lib/elixir/lib/map.ex

```diff
@@ -211,13 +220,16 @@ defmodule Map do
   """
   @spec take(map, [key]) :: map
   def take(map, keys) do
-    Enum.reduce(keys, [], fn key, acc ->
-      case fetch(map, key) do
-        {:ok, value} -> [{key, value} | acc]
-        :error -> acc
-      end
-    end)
-    |> :maps.from_list
+    do_take(keys, map, [])
+  end
```

Comment:
The following code

```
  def drop(map, list) when is_list(list) do
    drop_list(list, map)
  end
  def drop(map, enum) do
    enum
    |> Enum.reverse
    |> drop_list(map)
  end

  defp drop_list([], acc), do: acc
  defp drop_list([key | rest], acc) do
    drop_list(rest, Map.delete(acc, key))
  end
```

gets us the following improvement

```
Map.drop list manual recursion         1000   1886.05 µs/op
Map.drop list master                   1000   1937.43 µs/op
Map.drop range manual recursion        1000   2073.89 µs/op
Map.drop range master                  1000   2243.64 µs/op
```

worth? It's basically 10%.